crates/services/gas_price_service/src/v1/uninitialized_task.rs

```diff
@@ -0,0 +1,383 @@
+        let da_service = DaSourceService::new(self.da_source, poll_interval);
+
+        if BlockHeight::from(latest_block_height) == self.genesis_block_height
+            || first_run
+        {
+            let service = GasPriceServiceV1::new(
+                l2_block_source,
+                self.metadata_storage,
+                self.shared_algo,
+                self.algo_updater,
+                da_service,
+            );
+            Ok(service)
+        } else {
+            if latest_block_height > metadata_height {
+                sync_gas_price_db_with_on_chain_storage(
+                    &self.settings,
+                    &self.config,
+                    &mut self.metadata_storage,
+                    &self.on_chain_db,
+                    metadata_height,
+                    latest_block_height,
+                )?;
+            }
+
+            let service = GasPriceServiceV1::new(
+                l2_block_source,
+                self.metadata_storage,
+                self.shared_algo,
+                self.algo_updater,
+                da_service,
+            );
+            Ok(service)
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl<L2DataStore, L2DataStoreView, GasPriceStore, Metadata, DA, SettingsProvider>
+    RunnableService
+    for UninitializedTask<L2DataStoreView, GasPriceStore, Metadata, DA, SettingsProvider>
+where
+    L2DataStore: L2Data,
+    L2DataStoreView: AtomicView<LatestView = L2DataStore>,
+    GasPriceStore: GasPriceData,
+    Metadata: MetadataStorage,
+    DA: DaBlockCostsSource,
+    SettingsProvider: GasPriceSettingsProvider,
+{
+    const NAME: &'static str = "GasPriceServiceV0";
+    type SharedData = SharedV1Algorithm;
+    type Task = GasPriceServiceV1<FuelL2BlockSource<SettingsProvider>, Metadata, DA>;
+    type TaskParams = ();
+
+    fn shared_data(&self) -> Self::SharedData {
+        self.shared_algo.clone()
+    }
+
+    async fn into_task(
+        self,
+        _state_watcher: &StateWatcher,
+        _params: Self::TaskParams,
+    ) -> anyhow::Result<Self::Task> {
+        UninitializedTask::init(self)
+    }
+}
+
+pub fn initialize_algorithm<Metadata>(
+    config: &V1AlgorithmConfig,
+    latest_block_height: u32,
+    metadata_storage: &Metadata,
+) -> GasPriceResult<(AlgorithmUpdaterV1, SharedV1Algorithm)>
+where
+    Metadata: MetadataStorage,
+{
+    let algorithm_updater;
+    if let Some(updater_metadata) = metadata_storage
+        .get_metadata(&latest_block_height.into())
+        .map_err(|err| GasPriceError::CouldNotInitUpdater(anyhow::anyhow!(err)))?
+    {
+        let previous_metadata: V1Metadata = updater_metadata.try_into()?;
+        algorithm_updater = v1_algorithm_from_metadata(previous_metadata, config);
+    } else {
+        algorithm_updater = config.into();
+    }
+
+    let shared_algo =
+        SharedGasPriceAlgo::new_with_algorithm(algorithm_updater.algorithm());
+
+    Ok((algorithm_updater, shared_algo))
+}
```

Comment:
i think this is a dupe of `initialize_algorithm` from the v1/service.rs file, right?

Reply:
https://github.com/FuelLabs/fuel-core/pull/2442/files#diff-ddae082e091f4688309a88faa0e368835781b2edd78044fd50a06d0c7b7dd428L291